bluez-qt/bluetoothdevicemodel.cpp

```diff
@@ -176,14 +176,17 @@ void BluetoothDevicesModel::defaultAdapterChanged(QDBusObjectPath path)
 
 	if (adapter)
 	{
-		beginRemoveRows(QModelIndex(), 0, m_devices.size()-1);
-		foreach(BluetoothDevice* device, m_devices)
+		if (m_devices.size())
 		{
-			updateConnected(device->connected());
-			device->deleteLater();
+			beginRemoveRows(QModelIndex(), 0, m_devices.size()-1);
+			foreach(BluetoothDevice* device, m_devices)
+			{
+				updateConnected(device->connected());
+				device->deleteLater();
+			}
+			m_devices.clear();
+			endRemoveRows();
 		}
-		m_devices.clear();
-		endRemoveRows();
 
```

Comment:
Shouldn't updateConnected(false) be called here after we have removed all the devices from the list? Otherwise m_connected could remain true.